### ISSUE TITLE
[Subtask][Improvement][AQE] Introduce the new MergedShuffleDataSegment to reduce number of rpc

### ISSUE THREAD

**Issue by zuston** [apache/incubator-uniffle#302]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues) and found no similar issues.

### Describe the subtask

Currently, when using the `LocalOrderSegmentSplitter`, it will split multiple `ShuffleDataSegment` by three factors.

1. taskId in [startMapId, endMapId) taskIds bitmap
2. ShuffleDataSegment size should < readBufferSize
3. ShuffleDataSegment's blocks should be continuous

In some cases, these requirements will make the data size very small (<14M). So we could combine some ShuffleDataSegment to single one data-fetching request to reduce the RPC interaction number.

### Parent issue

#137 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by jerqi** [comment]
Maybe global order will be a better solution for this case.

**Comment by zuston** [comment]
Yes, maybe we should introduce global order to compare with other distribution types in different data size

but this pr is still meaningful for local order
